scratch/camera_feed/display_camera_feed.cpp

```diff
@@ -0,0 +1,202 @@
+}
+
+void setRemaining(unsigned char** byteBuffer, int section, uint8_t value){
+		*(((uint8_t*) (byteBuffer[section]))+3) = value;
+		return;
+}
+
+uint8_t getRemaining(unsigned char** byteBuffer, int section){
+		uint8_t remaining = *(((uint8_t*) (byteBuffer[section]))+3);
+		std::cout << "Remaining: " << remaining << std::endl;
+		return remaining;
+}
+
+void decreaseRemaining(unsigned char** byteBuffer, int section){
+		(*(((uint8_t*) (byteBuffer[section]))+3))--;
+		return;
+}
+
+int main(){
+		std::cout << "Starting" << std::endl;
+		//create a 2d array of buffers.
+		int pitch = 1920*3, *pitchptr = &pitch;
+		unsigned char** frame_buffer = new unsigned char*[CAMERA_FRAME_BUFFER_COUNT];
+		void* pixel_location;
+		//
+		for(int i = 0; i < CAMERA_FRAME_BUFFER_COUNT; i++){
+				frame_buffer[i] = new unsigned char[CAMERA_FRAME_BUFFER_SIZE];
+		}
+		uint16_t frame_list[CAMERA_FRAME_BUFFER_COUNT];
+		uint16_t data_size[CAMERA_FRAME_BUFFER_COUNT];
+		uint8_t indx = 0;
+		//establish a newtwork connection
+		network::Connection conn;
+		//establish network error
+		const char* destination_port = "127.0.0.1";
+		network::Error error = network::connect(&conn, destination_port, 45546, 45545) ;
+		if(error == network::Error::OK)
+				std::cout << "Error bound a port correctly" << std::endl;
+		else if(error == network::Error::OPEN_SOCKET)
+				std::cout << "Open socket error: " << std::endl;
+		else if(error == network::Error::BIND_SOCKET)
+				std::cout << "Binding socket error" << std::endl;
+		//initiate SDL, if its 0 something went wrong. Then we print the err
+		if (SDL_Init(SDL_INIT_VIDEO) != 0){
+				std::cout << "SDL_Init Error: " << SDL_GetError() << std::endl;
+				return 1;
+		}
+		//creates window. pramaeters are window title, x start pos, y start pos,
+		//x end pos, and y end pos.
+		SDL_Window *win = SDL_CreateWindow("Hello World!", 100, 100, 640, 480, SDL_WINDOW_SHOWN);
+		//check for failure
+		if (win == nullptr){
+				std::cout << "SDL_CreateWindow Error: " << SDL_GetError() << std::endl;
+				SDL_Quit();
+				return 1;
+		}
+		//makes a renderer. args are the window to render to, the index of the 
+		//rendering driver, and the rendering flags or'd together
+		//SDL ACCELERATED uses hardware acceleration, and SDL PRESENTVSYNC
+		//is present is synchronized with the refresh rate
+		//Other flags are TARGETTEXTURE -- supports redering to texture
+		// and SOFTWARE -- which means the renderer is a software fallback
+		SDL_Renderer *ren = SDL_CreateRenderer(win,-1,
+						SDL_RENDERER_ACCELERATED   | SDL_RENDERER_PRESENTVSYNC);
+		if (ren == nullptr){
+				SDL_DestroyWindow(win);
+				std::cout << "SDL_CreateRenderer Error: " << SDL_GetError() << std::endl;
+				SDL_Quit();
+				return 1;
+		}
+		SDL_Texture *texture = SDL_CreateTexture(ren, SDL_PIXELFORMAT_RGB24, SDL_TEXTUREACCESS_STREAMING, 1024, 768);
+		std::cout << "Initialized everything" << std::endl;
+		std::cout << "Starting Loops" << std::endl;
+
+		bool running = true;
```

Comment:
I can't tell if you're updating the state of this variable, but this is clear spot to use a `do while` clause :)

Reply:
Switched to using a do while, still takes a bit to x out of the window.